code/game/machinery/autolathe.dm

```diff
@@ -243,7 +251,13 @@
 		if(istype(O, /obj/item/disk/design_disk))
 			var/obj/item/disk/design_disk/D = O
 			if(D.blueprint)
-				if(!(D.blueprint.build_type & AUTOLATHE)) // otherwise, would silently fail in AddDesign2Known
+				var/datum/design/design = D.blueprint // READ ONLY!!
+
+				if(design.id in files.known_designs)
+					to_chat(user, "<span class='warning'>This design has already been loaded into the autolathe.</span>")
+					return 1
```

Comment:
(Not part of review)
I am questioning why this has so many `return 1` in situations where this wouldnt even be a `return TRUE` scenario. Not asking you to refactor it here, but it is confusing me.